feat(notifications): add toggle to show group chat start as incoming …

hek4ek opened the pull request:
…call

- Added a new toggle in NotificationSettings to display group chat start notifications as incoming calls
- Updated notification handling logic to respect the new setting
- Added localization keys for the toggle label

### Pull Request Checklist

- [x] I read the [contributing guide](https://github.com/element-hq/element-ios/blob/develop/CONTRIBUTING.md).
- [x] Pull request contains a [changelog label](https://github.com/element-hq/element-x-ios/blob/develop/CONTRIBUTING.md#changelog).

**UI changes have been tested with:**
- [x] iPhone and iPad simulators in portrait and landscape orientations.
- [x] Dark mode enabled and disabled.
- [x] Various sizes of dynamic type.
- [x] Voiceover enabled.

Signed-off-by: [hek4ek@gmail.com](mailto:hek4ek@gmail.com)

Enhancement  for: https://github.com/element-hq/element-meta/issues/2429

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/element-hq/element-x-ios?pullRequest=3945) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/element-hq/element-x-ios?pullRequest=3945) before we can accept your contribution.<br/><hr/>**Viktor Kumanenkov** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/element-hq/element-x-ios?pullRequest=3945) it.</sub>

hek4ek committed:
feat(notifications): add toggle to show group chat start as incoming call

- Added a new toggle in NotificationSettings to display group chat start notifications as incoming calls
- Updated notification handling logic to respect the new setting
- Added localization keys for the toggle label

alexander-potemkin commented:
@stefanceriu , it's quite an unexpected outcome, to be honest - the very same PR has been accepted by iOS team (@pixlwave to be more specific) earlier ([here](https://github.com/element-hq/element-ios/pull/7875)) and only rejected by Android team for one reason: Element 'Classic' was not accepting anything new at all.

I can't as well see any references to the necessity to get a prior approval on the [Contribution Guide](https://github.com/element-hq/element-x-ios/blob/develop/CONTRIBUTING.md), nor in [Pull Request guide](https://github.com/element-hq/element-android/blob/develop/docs/pull_request.md).

I have, however, raised a separate issue as per your request: https://github.com/element-hq/element-meta/issues/2802
How could a process of discussion with product team be held?

As per my understanding - and please, do correct me if I'm wrong or missing something, rings are an essential part of the group interaction, which is required for your biggest customers as well, unless you plan to keep that as a separate / paid feature of course?

stefanceriu commented:
> the very same PR has been accepted by iOS team (@pixlwave to be more specific) earlier

That's probably because legacy already had that `Ring for group calls`. For ElementCall on the other hand the notification type has been [explicitly defined on the rust side](https://github.com/matrix-org/matrix-rust-sdk/blob/5730f0e00e4a0e2f874e80c92bdc9429537a61ab/crates/matrix-sdk/src/room/mod.rs#L3151) at the request on the VoIP team. 

My thought process here is that if we go add a toggle for every single setting folks disagree on then we'll end up in a sea of configuration options we cannot possibly test or maintain properly. 

For that reason Element X is taking a different approach and letting the product/design/ux team lead what we do, how and why.

alexander-potemkin commented:
> > the very same PR has been accepted by iOS team (@pixlwave to be more specific) earlier
> 
> That's probably because legacy already had that `Ring for group calls`. For ElementCall on the other hand the notification type has been [explicitly defined on the rust side](https://github.com/matrix-org/matrix-rust-sdk/blob/5730f0e00e4a0e2f874e80c92bdc9429537a61ab/crates/matrix-sdk/src/room/mod.rs#L3151) at the request on the VoIP team.
> 
> My thought process here is that if we go add a toggle for every single setting folks disagree on then we'll end up in a sea of configuration options we cannot possibly test or maintain properly.
> 
> For that reason Element X is taking a different approach and letting the product/design/ux team lead what we do, how and why.

Thanks for diving me into the details! It starts making a bit of sense now. 

It [looks like](https://github.com/matrix-org/matrix-rust-sdk/blame/5730f0e00e4a0e2f874e80c92bdc9429537a61ab/crates/matrix-sdk/src/room/mod.rs#L3151) you are the person behind that commit, that restricted the calls for 1:1 calls only? If that's correct, would you mind to elaborate on the reasons behind that restriction, please?

stefanceriu commented:
It's been a while but I believe I took over and delivered a draft PR from the VoIP team. If I'm not mistaken the logic was that if you start a call in a room then it was probably scheduled by other means (i.e. calendar invites) and that you shouldn't bother the whole room, especially the ones that weren't invited. DMs are more ad-hoc.

The requirements are listed in https://github.com/element-hq/element-meta/issues/2385

alexander-potemkin commented:
Yeah, I can see my comments and my screenshot reposted by someone at 2385 :)

I understand the distraction argumentation, but, unless I'm missing something, I can't see anyone saying that group calls must not be able to ring a bell.

Personally I know two various approaches:
- let the ring bell for the group: since I'm part of the group, I might want to get that notification: i.e. - it's a Disaster Recovery group and I want to be on top of everything that is happening there - it is urgent and every minute matters;
- please, don't bother me, I'm working and if I'm part of the group, it doesn't mean I want it to buzz every other time someone hit the ring button.

The PR offered gives the user an option to choose and hence to use the messenger in more use cases. I understand - it's a bit awkward - it tries to work around the fact, that the user level preference lies deeply under the hood - in SDK.

Based on the links you've provided, would it be correct to assume that @manuroe is the person, making the decisions?

If not - how shall I process?
I mean - I hear you are saying that I shall raise an issue at meta project - which I did. But from my past experience, that kind of issues used to leave me nowhere - there are 1.4K of them - it's unpractical to expect anyone to take care of all of them. I would rather not have the efforts of @hek4ek be left in vain. I can also see a demand for the feature from the users, which makes me believe that probably that shall be adjusted at Element side.

alexander-potemkin commented:
A gentle reminder

alexander-potemkin commented:
@stefanceriu , it is like 2 months passed, with no responce from none of the tickets - nor here, nor at [meta project issue](https://github.com/element-hq/element-meta/issues/2802) raised.
From checking other tickets, it feels like _any_ decision shall pass via product team and I've found [an issue](https://github.com/element-hq/element-web/issues/2042) that are 9 years old now - still pending product managers decision, making the use of the product extremely complicated for a regular user.

I understand that the product in use by a huge paying customers might have different priorities and that makes sense, so I understand that fork might be a better solution here.

May I kindly ask you to help with one thing only? We've tried changing the line in SDK, that filters out group calls, but the client still didn't raise a call for the group call.
**It would be of great help if you could confirm**, as per your memory, if SDK is the only place where the group calls are suppressed? Or shall some / any client side modifications be done as well?

I'm trying to make a fork for SDK and client with that setting altered and publish this to the stores - guess, it might ease some pressure (if any) on you and you could consider that as a tests polygon.

Thank you in advance in any way!

alexander-potemkin commented:
I understand that tagging everyone would not help (and might even hurt), but I don't know who is the right person to approach. I see that @pixlwave is doing lots of the releases, alongside with Stefan, hope Doug wouldn't mind my gentle ping in here.

alexander-potemkin commented:
> Fundamentally this isn't our choice to change the behaviour. Your best bet would be to discuss this with the Element Call team over in [#webrtc:matrix.org](https://matrix.to/#/#webrtc:matrix.org) (it's possible they didn't see your ticket in /meta) 🙂

Thank you very much, Doug - you're always of great help!
Posted, it will work better this time :)

alexander-potemkin commented:
> The thread concluded with the suggestion to take the discussion to the Element Call team on Matrix at #webrtc:matrix.org. @alexander-potemkin, were you able to get any new information or feedback from the team in that channel? Any update would be appreciated!

Hey, kind of - as per my understanding group calls 'ringing' nor planned nor will be implemented / accepted as a feature of the app anytime soon.

RebaseYourExpectations commented:
Hey, I totally agree with the comments you made about this behavior. What's currently implemented is more of a video conferencing feature than true group calls, and if the aim is for a true VoIP alternative, this really has to be revisited.

RebaseYourExpectations commented:
> > The thread concluded with the suggestion to take the discussion to the Element Call team on Matrix at #webrtc:matrix.org. @alexander-potemkin, were you able to get any new information or feedback from the team in that channel? Any update would be appreciated!
> 
> Hey, kind of - as per my understanding group calls 'ringing' nor planned nor will be implemented / accepted as a feature of the app anytime soon.

Hey, I totally agree with the comments you made about this behavior. What's currently implemented is more of a video conferencing feature than true group calls, and if the aim is for a true VoIP alternative, this really has to be revisited.
